lib/auth.js

```diff
@@ -231,6 +234,47 @@ module.exports = new Model({
     }.bind(this));
   },
 
+  signInWithOAuthRedirect: function (args, window) {
+    // Check to see if we're running in the browser or not
+    if (typeof window === 'undefined') {
+      throw new Error('OAuth login is only available via the browser.')
+    } else if (!args.appId) {
+      throw new Error('You need to specify an app ID.');
+    } else if (!args.redirectUri) {
+      throw new Error('You need to specify a URI to redirect to after authorization.');
+    }
+
+    var originalArguments = arguments;
+    return this.checkCurrent().then(function(user) {
+      if (user) {
+        return this.signOut().then(function() {
+          return this.signIn.apply(this, originalArguments);
+        }.bind(this));
+      } else {
+        console.log('Signing in with OAuth (redirect)');
+        var url = [
+          config.host,
+          '/oauth/authorize',
+          '?response_type=token',
+          '&client_id=',
+          args.appId,
+          '&redirect_uri=',
+          args.redirectUri
+        ].join('');
+        window.location.href = url;
+      }
+    }.bind(this));
+  },
+
+  completeSignInWithOAuth: function (args) {
```

Comment:
So - after you finish signing in on the Zoo login page, it then redirects you back to your app. But it needs to deserialize the response (which is supposed to be passed as a set of URL query parameters, but in our case it leaves off the leading `?`, yay). 

I couldn't think of a way of handling that which doesn't involve the app - so my current solution is to look for a route that contains those params, and call this function which completes the sign in process.